cloudformation/stacks/AutoSpotting/template.yaml

```diff
@@ -376,8 +381,8 @@
               Ref: "TerminationNotificationAction"
             PATCH_BEANSTALK_USERDATA:
               Ref: "PatchBeanstalkUserdata"
-            LAMBDA_MANAGE_ASG:
-              Ref: "LambdaManageASG"
+            SQS_QUEUE_URL:
```

Comment:
Is this queue now compulsory or can AutoSpotting still run in cron mode without this new FIFO queue?

I know of a few people who run it on Kubernetes from the Docker container images, and they probably can't easily provision an SQS queue.

Reply:
The only way to avoid any concurrency has been to handle all ondemand replacement by the sqs fifo queue.
So now cron events use the queue too.
If we do not do it this way there could be a probability, low but does exist, that two AutoSpotting Lambda invocation, one triggered by sqs queue and one triggered by cron schedule, act in same time on the same ASG.

But people running AutoSpotting in docker images, how are able to invoke LambdaManageASG in the current implementation ?

We can think about a flag, if not using the sqs queue we can in `replaceOnDemandInstanceWithSpot` execute internally swapWithGroupMember instead of sending the message to the sqsqueue.
So to keep only one method for replacing ondemand.
But probably we should need to make some changes to the code.
It's just an idea.

Reply:
Good point, I haven't seen any bug reports about this but it's probably broken. There are few people who use the Docker images though, so if there are any reports from them we can get this done later, I don't want to overcomplicate this PR as it's already quite complex

Reply:
@mello7tre how hard would it be to make this depend on the presence of the SQS queue in the configuration and if there's no queue defined just handle it without the queue?

Reply:
i would like to keep a single method `swapWithGroupMember` to replace ondemand, so need to look how to change code to use it directly, if conf.SQSQueueURL is empty, from current cron code.
Currently i am working on https://github.com/AutoSpotting/AutoSpotting/pull/458#discussion_r621955442.
When that part is stable i will give this a look.